diff -ruN sox-12.17.9-old/src/sox.c sox-12.17.9/src/sox.c
--- sox-12.17.9-old/src/sox.c	2005-12-15 12:39:36.000000000 +0100
+++ sox-12.17.9/src/sox.c	2005-12-15 12:42:59.000000000 +0100
@@ -1111,7 +1111,11 @@
                              &efftab[neffects-1].obuf[total],
                              (st_ssize_t)efftab[neffects-1].olen-total);
 
-              if (len < 0)
+	      /* GG: The eof test might not be necessary, but I am 
+	       * paranoid here and don't have the time to check if
+	       * out of space behaviour has been fixed */
+	      
+              if (len < 0 || file_desc[file_count-1]->file.eof)
               {
                   st_warn("Error writing: %s",
                           file_desc[file_count-1]->st_errstr);
